### DISCUSSION TITLE
Embedding (source 1), ChatGPT (source 2) - How?

### DISCUSSION THREAD

**Discussion by josetamayo5** [FlowiseAI/Flowise#587] [Q&A]
Hi! I need some help.

I'm starting to use Flowise and I think it's a fantastic tool. I have created a chatbot connected to a Pinecone embedding.

I would like that when the chatbot doesn't find an answer in the embedding (source 1), it makes use of the ChatGPT intelligence (source 2).

But always prioritising the embedding.

How can I do this?

I would really appreciate a help, a screenshot....

Thanks :-)

**Comment by HenryHengZJ** [comment]
We had a POC that uses embeddings score values to determine whether or not to use embeddings or chatgpt intelligence to build the reply. If embeddings score is lower than certain thershold, use chatgpt.

From our testing, it seems like we can't really use that score as deterministic factor.

As you can see from pic below, I tried asking `hello my name is ben`, I expected the embeddings score should be low but in fact its ~0.78
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/d8722a84-7a0e-4da6-80ac-18e5ec2b41e0)

Then when I tried to ask a relevant question: `where are the available locations`, the score are lower ~0.75
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/013eab55-933c-4a4f-ac2f-9a46b177d9d0)
